Notes/BeetleBattleDesign.md

```diff
@@ -0,0 +1,115 @@
+
+So the proposal is that we keep track of two scalar floats of speed, `forward_speed` and `sideways_speed`. Note that
+these are scalar, not vector values, but that's OK because so are the sprite attributes. Essentially, we will convert
+these speed values by finding the global x and y components of the vectors they describe based on `angle`. It might be
+useful to show this process on a diagram:
+
+![A diagram of our calculations of the global x and y components of our forward and sideways vectors.](BeetleSpriteVectors.png)
+
+While `forward_speed` and `sideways_speed` are scalars, the vectors they describe are drawn in green. We will have to
+calculate the purple vectors that are the x and y components. Luckily, we can use basic trigonometry here. Take notice
+of θ, which is 90 - `angle`. This makes sin(θ) equal to the length of the y-component of the forward vector divided
+the length of the length of the forwards vector. We can reason likewise for the x-component and cos(θ). Putting this
+together, and realizing that if θ is 90 - `angle`, then it is the other angle in the triangle, and thus we can just use
+`angle` directly:
+
+> `forward_x` = sin(`angle`) \* `forward_speed`
+> `forward_y` = cos(`angle`) \* `forward speed`
+> `sideways_x` = cos(`angle`) \* `sideways_speed`
+> `sideways_y` = sin(`angle`) \* `sideways_speed`
+> `change_x` = `forward_x` + `sideways_x`
+> `change_y` = `forward_y` + `sideways_y`
+
+Thus this is some pseudo-code for handling movement.
+
+### The Projectiles
+
+The projectiles are shot along the facing direction at regular intervals. There is likely a cooldown, the shot is fired,
+and then a bullet particle is shot. If this bullet particle collides with an enemy beetle it does damage. Note, I say "enemy" beetle because I don't think we want friendly fire between the beetles.
+
+These are the weakest attacks, made up by their continuous firing. Stronger attacks exist and are rewarded by beetles
+acquiring traits.
+
+### The Teams
+
+The beetles are divided into two battling teams that start on opposite sides. By default there are three beetles in a
+team, though team traits gained through evolution can add to that number. By default all the beetles in a team are
+equal, though certain team traits may specialize a single beetle with a particularly powerful ability.
+
+The battle ends when all the beetles on either team have been eliminated, making the other team the victor. It could
+technically be possible for the two teams to be eliminated at the same time, but most likely one team over the other
+would "win" based on when processing for each beetle occurs and when victory is awarded.
+
+### The Traits
+
+Each team has a set of traits. These traits apply to the beetles to change their capabilities. These traits may affect
+the entire team of beetles as a whole, give a certain trait to all the beetles in a team, or give a certain trait to
+a single beetle in a team.
```

Comment:
This paragraph is broadly true, and we have flexibility in how we want to apply effects to our beetles, esp with the changes we'll be making.

But for elucidation, in the original, a mutation applies to either all beetles, or only elites (queens are a special type of non-attacking elite, so elite weapons do not affect queens but e.g. elite armor or other bonuses do).